provider/sgx.py

```diff
@@ -0,0 +1,189 @@
+"""
+Module for sgx relevant operations.
+"""
+import logging
+import re
+
+from avocado.utils import process
+from virttest.staging import utils_memory
+from virttest.utils_misc import normalize_data_size
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class SGXError(Exception):
+    """ General SGX error"""
+    pass
+
+
+def _get_epc_size(output):
+    """
+    Get epc size from system message
+
+    :param output: epc section info
+    :return: total epc size
+    """
+    epc_size = 0
+    for line in output.splitlines():
+        tmp_epc = re.search(r'\b0x[0-9a-fA-F]*-0x[0-9a-fA-F]*\b',
+                            str(line))
+        if tmp_epc:
+            epc_size += int(tmp_epc[0].split('-')[1], 16) - int(
+                tmp_epc[0].split("-")[0], 16) + 1
+    return epc_size
+
+
+class SGXTestCapability(object):
+    """
+    Hypervisor sgx test capabilities check.
+    """
+
+    def __init__(self, test, params):
+        """
+
+        :param test: Context of test.
+        :param params: params of running ENV.
+        """
+        self._test = test
+        self._params = params
+
+    def valid_host_sgx_cap(self):
+        """
+        Valid if host sgx enablement
+
+        :return: host epc size/False
+        """
+        try:
+            host_sgx_msg = process.system_output(
+                self._params["host_sgx_check"], shell=True)
+        except Exception as e:
+            self._test.cancel("Host sgx capability check fail %s" % e)
+        else:
+            host_msg_epc_size = _get_epc_size(host_sgx_msg)
+            return host_msg_epc_size
+
+    def valid_host_numa_node(self):
+        """
+        Valid if host numa nodes satisfy test requirement
+
+        :return: True/False
+        """
+        LOG_JOB.debug("Check host's numa node(s)!")
+        host_numa = utils_memory.numa_nodes()
+        node_list = []
+        numa_info = process.getoutput("numactl -H")
+        for i in host_numa:
+            node_size = re.findall(r"node %d size: "
+                                   r"\d+ \w" % i, numa_info)[0].split()[-2]
+            if node_size != '0':
+                node_list.append(str(i))
+        if len(node_list) < int(self._params["monitor_expect_nodes"]):
```

Comment:
it may be better to pass in monitor_expect_nodes as an argument of this function, but yours is also OK for me

Reply:
Thanks! Updated.